File: Python/17-longest-palindrome.py
def is_palindrome(str):
    i = 0

    while i < len(str)/2:
        if str[i] != str[-i-1]:
            return False

        i += 1

    return True

def longest_palindrome(str):
    result = ""
    length = 0

    for i in range(0, len(str)):
        for j in range(i+1, len(str)+1):
            substr = str[i:j]
            if is_palindrome(substr) and \
               len(substr) > length:
                result = substr
                length = len(substr)


    return result

print(longest_palindrome("abcbedeffe"))

File: Javascript/17-longest-palindrome.js
function palindrome(string) {
	var len = string.length;

	for (var i = 0; i < Math.floor(len/2); i++) {
		if (string[i] != string[len - 1 - i]) {
			return false;
		}
	}

	return true;
}

function longestPalindrome(string) {
	var substring = "";
	var result = "";

	if (string.length == 1) {
		return string;
	}

	for (var i = 0; i < string.length-2; i++) {
		for (var j = 2; j < string.length; j++) {
			substring = string.slice(i,j+1);
			if (palindrome(substring) &&
				 substring.length > result.length) {
				result = substring;
			}
		}
	}

	return result;
}

console.log(longestPalindrome("a"));
